File: src/Services/Book/Book.Application/DTOs/Author/RequestDTOs/UpdateAuthorDto.cs
namespace Book.Application.DTOs.Author.RequestDTOs;

public sealed record UpdateAuthorDto(
    int Id,
    string? OpenLibraryKey,
    string FirstName,
    string LastName,
    string FullName,
    DateOnly? BirthDate,
    DateOnly? DeathDate,
    string? Biography,
    string? ImageUrl);

File: src/Client/Angular.Client/src/app/models/author/updateAuthorDto.ts
export interface UpdateAuthorDto {
    id: number;
    openLibraryKey: string | null;
    firstName: string;
    lastName: string;
    fullName: string;
    birthDate: string;
    deathDate: string | null;
    biography: string | null;
    imageUrl: string | null;
}
